### PR TITLE
fix(ui5-application-inquirer): Apply defaults. TS default prompt fix.

### PR DESCRIPTION
**Pull request by IainSAP**
#2034 

- Apply defaults to not executed prompts (advanced options only)
- Ensure `EnableTypeScript` default option should be a function that is passed answers + cds info
- Updates tests
-  fix some review comments (typos) raised here: https://github.com/SAP/open-ux-tools/pull/1986

### CODE REVIEW

File: packages/inquirer-common/src/prompts/helpers.ts [hunk 1]
```diff
@@ -123,7 +136,7 @@ export function extendWithOptions<T extends YUIQuestion = YUIQuestion>(
 
             for (const extProp of propsToExtend) {
                 if (extProp === 'validate' || extProp === 'additionalMessages') {
-                    question = applyExtensionFunction(question, promptOpt as CommonPromptOptions, extProp);
+                    question = applyExtensionFunction(question, promptOpt as CommonPromptOptions, extProp, promptState);
                 }
                 // Provided defaults will override built in defaults
                 const defaultOverride = (promptOptions[promptOptKey] as PromptDefaultValue<string | boolean>).default;
```
Comment (IainSAP) [inline packages/inquirer-common/src/prompts/helpers.ts:142]: Take care of default functions!

**Comment by IainSAP** [comment]
Closing, only advanced option defaults should be applied not all unanswered prompts...

**Review by devinea — APPROVED** [review]
Thanks @IainSAP. 
changeset ✅ 
Code changes are expected. 
Tests and snapshot updates seem appropriate. 
I haven't tested locally.

**Review by cianmSAP — DISMISSED** [review]
Code updates look good
Tests to cover
Changeset ✅

**Review by cianmSAP — APPROVED** [review]
Re-approving

**Review by devinea — APPROVED** [review]
Re-approving

**Pull request merged**